portal/views/home_new.py

```diff
@@ -0,0 +1,158 @@
+# authorisation of Ocado.
+#
+# Any propagation, distribution or conveyance of this program must include this
+# copyright notice and these terms. You must not misrepresent the origins of this
+# program; modified versions of the program must be marked as such and not
+# identified as the original program.
+from functools import partial
+
+from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse, reverse_lazy
+from django.contrib import messages as messages
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.forms import AuthenticationForm
+from recaptcha import RecaptchaClient
+from django_recaptcha_field import create_form_subclass_with_recaptcha
+
+from portal.forms.teach import TeacherLoginForm
+from portal.helpers.emails import send_verification_email, is_verified
+from portal.utils import using_two_factor
+from portal import app_settings
+from ratelimit.decorators import ratelimit
+
+recaptcha_client = RecaptchaClient(app_settings.RECAPTCHA_PRIVATE_KEY, app_settings.RECAPTCHA_PUBLIC_KEY)
+
+
+def teach_email_labeller(request):
+    if request.method == 'POST' and 'login' in request.POST:
+        return request.POST['login-email']
+
+    return ''
+
+
+def login_popup(request):
+    return render_forms(request, 'redesign/login_popup.html')
+
+
+def home_new(request):
+    return render_forms(request, 'redesign/home_new.html')
+
+
+def teach_new(request):
+    return render_forms(request, 'redesign/teach_new.html')
+
+
+def play_new(request):
+    return render_forms(request, 'redesign/play_new.html')
+
+
+def about_new(request):
+    return render_forms(request, 'redesign/about_new.html')
+
+
+def help_and_support_new(request):
+    return render_forms(request, 'redesign/help-and-support_new.html')
+
+
+def terms_new(request):
+    return render_forms(request, 'redesign/terms_new.html')
+
+
+@ratelimit('ip', periods=['1m'], increment=lambda req, res: hasattr(res, 'count') and res.count)
+@ratelimit('email', labeller=teach_email_labeller, ip=False, periods=['1m'], increment=lambda req,
+           res: hasattr(res, 'count') and res.count)
+def render_forms(request, rendered_url):
+    invalid_form = False
+
+    limits = getattr(request, 'limits', {'ip': [0], 'email': [0]})
+    captcha_limit = 5
+
+    using_captcha = (limits['ip'][0] > captcha_limit or limits['email'][0] > captcha_limit)
+    should_use_captcha = (limits['ip'][0] >= captcha_limit or limits['email'][0] >= captcha_limit)
+
+    LoginFormWithCaptcha = partial(
+        create_form_subclass_with_recaptcha(TeacherLoginForm, recaptcha_client), request)
+    InputLoginForm = LoginFormWithCaptcha if using_captcha else TeacherLoginForm
+    OutputLoginForm = LoginFormWithCaptcha if should_use_captcha else TeacherLoginForm
+
+    login_form = OutputLoginForm(prefix='login')
+
+    if request.method == 'POST':
```

Comment:
Code Climate has flagged this function up as having very high cyclomatic complexity.  I'm new to Code Climate, but I suspect the problem is this little cascade of nested `if`s right here.

Refactor this bit by taking your deepest-nested `if/else`, and turning it into a function of its own.  Then work your way up the stack until you only have only one layer, and no nesting.  

It will help you a lot if you have test coverage of this already.  If you don't, now's the time to write some tests, preferably before you start to refactor.